### ISSUE TITLE
alias for kakoune's value expensions

### ISSUE THREAD

**Issue by sucrecacao** [mawww/kakoune#3794]
### Feature 

Being able to use alias for kakoune `val` variable:
`alias %val{sid}=%val{main_reg_hash}`

Using `option` wouldn't work, because it would just store the current state of the `val`. An alias is just a shortname for the value expensions.

### Usecase 

When filter with the `$` command, entering $kak_main_reg_hash is so long it might be fast to do the filtering by hand. The name is also so long  one most probably forgot the exact orthography. 
This is a very particular example, but I'm sure aliasing could have a lot of other usecases.

**Comment by krobelus** [comment]
A workaround is `set-register x %{$kak_main_reg_hash}`

**Comment by sucrecacao** [comment]
@krobelus , Thx
Didn't succeed to make it work. Are you sure you don't have a typo in your example ?
Also, should I call the register afterward ?

**Comment by krobelus** [comment]
Sorry that wasn't clear, this is just as an abbreviation: use `<c-r>x` in the prompt instead of typing the full thing.
Can be done with `map global prompt ...` as well.

**Comment by alexherbo2** [comment]
@sucrecacao You can create a prompt mapping to type that value.

``` kak
map -docstring 'Type $kak_main_reg_hash' global prompt <a-i> '$kak_main_reg_hash'
```

@krobelus It will not work, because you expand to the value of the main selection in the prompt, instead of typing the variable.

**Comment by alexherbo2** [comment]
In [connect.kak], I have a `connect_environment` option to configure the shell environment.

[connect.kak]: https://github.com/alexherbo2/connect.kak

It could be nice to have a similar functionality for the shell commands.

``` kak
set-option global shell_environment %{
  i=$kak_main_reg_hash
}
```

So you can refer to `$i` as `$kak_main_reg_hash` in the `$` command.

Create custom functions, etc.

**Comment by krobelus** [comment]
> It will not work, because you expand to the value for the main selection in the prompt, instead of typing the variable.

Yes it does because I quoted it: after `set-register x %{$kak_main_reg_hash}`, typing `!echo <c-r>x` will result in a prompt with `!echo $kak_main_reg_hash`. Though I agree that a prompt mapping is better for this use case.

**Comment by alexherbo2** [comment]
Oh right.  It was not clear the value was literal when I see `%{ ... }` instead of single quoted string.
